src/OrchardCore.Modules/OrchardCore.Workflows/Controllers/WorkflowTypeController.cs

```diff
@@ -549,5 +574,31 @@ public async Task<IActionResult> Delete(long id)
 
             return activityShape;
         }
+
+        private async Task<string> ExportWorkflows(params long[] itemIds)
+        {
+            using var fileBuilder = new TemporaryFileBuilder();
+            var archiveFileName = fileBuilder.Folder + ".zip";
+            var recipeDescriptor = new RecipeDescriptor();
+            var deploymentPlanResult = new DeploymentPlanResult(fileBuilder, recipeDescriptor);
+            var data = new JArray();
+            deploymentPlanResult.Steps.Add(new JObject(
+                new JProperty("name", "WorkflowType"),
+                new JProperty("data", data)
+            ));
+            //Do filter
+            foreach (var workflow in await _workflowTypeStore.GetAsync(itemIds))
+            {
+                var objectData = JObject.FromObject(workflow);
+                // Don't serialize the Id as it could be interpreted as an updated object when added back to YesSql
+                objectData.Remove(nameof(workflow.Id));
+                data.Add(objectData);
+            }
+
+            await deploymentPlanResult.FinalizeAsync();
```

Comment:
Use `IDeploymentManager.ExecuteDeploymentPlanAsync()` instead.

Reply:
Here is to be derived from the system workflow type, why to execute `ExecuteDeploymentPlanAsync`.

Reply:
The code in the controller duplicates code from there. More importantly, it spreads the export logic around, which makes later changes harder and more error-prone.

Reply:
Done